### PR TITLE
SSCS-7515 SSCS-7516 Refactor to support WriteFinalDecision for different benefit types - PR1

### PR DESCRIPTION
**Pull request by michaellavelle**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SSCS-7515
https://tools.hmcts.net/jira/browse/SSCS-7516

### Change description ###

Refactor to support WriteFinalDecision for different benefit types

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/writefinaldecision/pip/PipAwardType.java [hunk 1]
```diff
@@ -1,14 +1,14 @@
-package uk.gov.hmcts.reform.sscs.ccd.presubmit.writefinaldecision;
+package uk.gov.hmcts.reform.sscs.ccd.presubmit.writefinaldecision.pip;
 
 /**
  * Enum specifying the possible types of Award, along with the key for that award in CCD.
  */
-public enum AwardType {
+public enum PipAwardType {
```
Comment (jackmaloney) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/writefinaldecision/pip/PipAwardType.java:6]: Is this not consistent across benefit types? How different are they?

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/writefinaldecision/WriteFinalDecisionAboutToSubmitHandler.java [hunk 2]
```diff
@@ -59,43 +59,29 @@ public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callb
...
 
-    private List<String> getDecisionNoticePointsValidationErrorMessages(SscsCaseData sscsCaseData) {
-
-        return Arrays.stream(PointsCondition.values())
-            .filter(pointsCondition -> pointsCondition.isApplicable(sscsCaseData))
-            .map(pointsCondition ->
-                getOptionalErrorMessage(pointsCondition, sscsCaseData))
+    private <T extends PointsCondition<?>> List<String> getDecisionNoticePointsValidationErrorMessages(Class<T> enumType, DecisionNoticeQuestionService decisionNoticeQuestionService, SscsCaseData sscsCaseData) {
+        return Arrays.stream(enumType.getEnumConstants())
+            .filter(pipPointsCondition -> pipPointsCondition.isApplicable(sscsCaseData))
```
Comment (jackmaloney) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/writefinaldecision/WriteFinalDecisionAboutToSubmitHandler.java:80]: very minor point but should `pipPointsCondition` be named something more generic?

**Review by jackmaloney — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/writefinaldecision/pip/PipAwardType.java (hunk 1 above)
Comment (michaellavelle) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/writefinaldecision/pip/PipAwardType.java]: I've now merged these award types into a single enum

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/writefinaldecision/WriteFinalDecisionAboutToSubmitHandler.java (hunk 2 above)
Comment (michaellavelle) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/writefinaldecision/WriteFinalDecisionAboutToSubmitHandler.java]: I've now renamed to pointsCondition

**Pull request merged**